cranelift-codegen/src/context.rs

```diff
@@ -49,6 +50,9 @@ pub struct Context {
 
     /// Loop analysis of `func`.
     pub loop_analysis: LoopAnalysis,
+
+    /// Peephole optimiser context.
+    pub peephole: Peephole,
```

Comment:
As suggested by @lars-t-hansen, can this live in the register allocation context, and can the peephole code be moved within this directory too? It would make sense to do so since the pass is only needed because of the register allocation's shortcomings.

Reply:
The discussion yesterday at first suggested that we need to do the reload optimization after prologue/epilogue since the stack layout happens in that phase, but then later we got around to the point that it's just the identity of the stack slot that matters, and that's available sooner, so I too would be interested in seeing further investigation here to move this pass into regalloc.